### PR TITLE
added the plot functionality for modelling data and plot images are being saved in vevestaXDump folder

### PR DESCRIPTION
**Pull request by sparsh459**

### CODE REVIEW

File: vevestaX/vevesta.py [hunk 1]
```diff
@@ -201,91 +202,66 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
         
         # excluding numeric, datetime type
         nonNumericColumns = modelingData.select_dtypes(exclude=['number', 'datetime'])
         modelingData.drop(nonNumericColumns, axis=1, inplace=True)
         modelingData.drop('experimentID', axis=1, inplace=True)
-        print(modelingData)
-        # print(nonNumericColumns)
+        
+        # changing the values in timestamp column to dates (vector slicing)
+        nonNumericColumns['timestamp in UTC'] = nonNumericColumns['timestamp in UTC'].str[:9]
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:228]: Dont use str to extract date, use pandas.to_datetime(['timestamp in UTC'] with right timestamp format.

File: vevestaX/vevesta.py [hunk 2]
```diff
@@ -201,91 +202,66 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+
+        # creating a new folder in current directory
+        Path("vevestaXDump").mkdir(parents=True, exist_ok=True)
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:231]: create a variable name with folder name and then use that...may be create it as a property of class vevesta. Also truncate the existing contents inside the folder before writing the new contents (images).

File: vevestaX/vevesta.py [hunk 3]
```diff
@@ -201,91 +202,66 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
 
         workbbok = openpyxl.load_workbook(filename)
         workbbok.create_sheet('performancePlots')
-        # workbook = xlsxwriter.Workbook(filename)
         plotSheet=workbbok['performancePlots']
         xAxis = list(nonNumericColumns['timestamp in UTC'])
-        
-        columnValue = 4
+        img_dir = 'vevestaXDump/'
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:237]: u have used vevestaX dump in multiple places. Use a variable.

File: vevestaX/vevesta.py [hunk 4]
```diff
@@ -201,91 +202,66 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+            plt.savefig(os.path.join(img_dir,imagename), bbox_inches='tight') 
 
+            img = openpyxl.drawing.image.Image(os.path.join(img_dir,imagename))  
             img.anchor = columntext
             plotSheet.add_image(img)
 
             columnValue+=20
 
         workbbok.save(filename)
-        # workbook.close()
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:277]: Close the work book after all figures have been written to the excel.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 5]
```diff
@@ -1,14 +1,16 @@
+from email.utils import localtime
 import pandas
 import inspect
 import os.path
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:4]: you have imported os.path and os both, remove os.path from imports

File: vevestaX/vevesta.py [hunk 6]
```diff
@@ -201,91 +204,87 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
-        # skipcols = ['experimentID', 'features']
-        # cols = for column in modelingData
-
-        if (os.path.isfile(filename)):
-            modelingData = pandas.read_excel(filename, sheet_name='modelling', index_col=[])
-
-        modelingData['timestamp in UTC'] = pandas.to_datetime(['timestamp in UTC'],)
-        print(modelingData['timestamp in UTC'].dtypes)
-
+        if (os.path.isfile(fileName)):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:221]: you can write this whole logic in a function, __getModellingDataFromExcel(filename).

Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:221]: Check if the file exist, if the file doesnt exist then return and do nothing

File: vevestaX/vevesta.py [hunk 7]
```diff
@@ -201,91 +204,87 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+        if len(modelingData.columns) == 0:
+            return 
         
-        columnValue = 4
+        directoryToDumpData = 'vevestaXDump'
+        self.__truncateFolder(directoryToDumpData)
+        # creating a new folder in current directory
+        Path(directoryToDumpData).mkdir(parents=True, exist_ok=True)
+
+        workBook = openpyxl.load_workbook(fileName)
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:243]: check if the file exist, if it doesn't then create the workbook.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 8]
```diff
@@ -233,47 +233,47 @@ def __plot(self, filename):
...
             plotSheet.add_image(img)
 
             columnValue+=20
 
-        workbook.save(filename)
+        workBook.save(fileName)
 
     # to turncate teh content inside the vevestaXDump folder if it exist
-    def __emptfldr(self, dir_name):
+    def __truncateFolder(self, dir_name):
```
[the remark this answers is not in the document]
Comment (sparsh459) [reply vevestaX/vevesta.py]: Done in the new commit

File: vevestaX/vevesta.py [hunk 9]
```diff
@@ -233,47 +233,47 @@ def __plot(self, filename):
...
         folder = dir_name
```
[the remark this answers is not in the document]
Comment (sparsh459) [reply vevestaX/vevesta.py]: Done in the latest commit

File: vevestaX/vevesta.py (hunk 5 above)
Comment (sparsh459) [reply vevestaX/vevesta.py]: resolved

File: vevestaX/vevesta.py (hunk 6 above)
Comment (sparsh459) [reply vevestaX/vevesta.py:223]: done

Comment (sparsh459) [reply vevestaX/vevesta.py:223]: done

File: vevestaX/vevesta.py [hunk 10]
```diff
@@ -201,92 +203,95 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
-        # cols = for column in modelingData
-
-        if (os.path.isfile(filename)):
-            modelingData = pandas.read_excel(filename, sheet_name='modelling', index_col=[])
-
-        modelingData['timestamp in UTC'] = pandas.to_datetime(['timestamp in UTC'],)
-        print(modelingData['timestamp in UTC'].dtypes)
-
+        # checks if file exist then only if fetches modelling data
+        if (os.path.isfile(fileName)):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:223]: this should be part of the function self.__getModellignSheetData(fileName)

File: vevestaX/vevesta.py [hunk 11]
```diff
@@ -201,92 +203,95 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+            modelingData = self.__getModellignSheetData(fileName)
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:224]: after this function, check if modelingData is empty, then return.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 12]
```diff
@@ -201,92 +203,95 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+
+                img = openpyxl.drawing.image.Image(os.path.join(directoryToDumpData,imageName))  
+                img.anchor = columnText
+                plotSheet.add_image(img)
+
+                columnValue+=20
+
+            workBook.save(fileName)
+
+    # to turncate teh content inside the vevestaXDump folder if it exist
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:274]: correct the spelling of teh to the

File: vevestaX/vevesta.py [hunk 13]
```diff
@@ -201,92 +203,95 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+
+                plt.savefig(os.path.join(directoryToDumpData,imageName), bbox_inches='tight') 
+
+                img = openpyxl.drawing.image.Image(os.path.join(directoryToDumpData,imageName))  
+                img.anchor = columnText
+                plotSheet.add_image(img)
+
+                columnValue+=20
+
+            workBook.save(fileName)
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:272]: after all the iterations of the plot you need to close the workbook explicitly.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py (hunk 12 above)
Comment (Priyanka-Vevesta) [reply vevestaX/vevesta.py:274]: and truncate not turncate

File: vevestaX/vevesta.py [hunk 14]
```diff
@@ -124,11 +125,12 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
         featureEngineeringData = None
         messageData = None
         experimentID = 1
+        localTimestamp = localtime().strftime("%Y-%m-%d %H:%M:%S")
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:128]: use datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")

File: vevestaX/vevesta.py [hunk 15]
```diff
@@ -1,14 +1,15 @@
+from email.utils import localtime
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:1]: remove this

File: vevestaX/vevesta.py [hunk 16]
```diff
@@ -1,14 +1,15 @@
...
 import pandas
 import inspect
-import os.path
 import ipynbname
-from datetime import datetime
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:5]: use this...dont remove it

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 17]
```diff
@@ -201,92 +203,102 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+                
+                img = openpyxl.drawing.image.Image(os.path.join(directoryToDumpData,imageName))  
+                img.anchor = columnText
+                plotSheet.add_image(img)
+                
+
+                columnValue+=20
+
+            workBook.save(fileName)
+            workBook.close()
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:277]: i dont think you have indented it right. Did u test the code ?

File: vevestaX/vevesta.py [hunk 18]
```diff
@@ -201,92 +203,102 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+                try:
+                    if os.path.isfile(file_path) or os.path.islink(file_path):
+                        os.unlink(file_path)
+                    elif os.path.isdir(file_path):
+                        shutil.rmtree(file_path)
+                except Exception as e:
+                    print('Failed to delete %s. Reason: %s' % (file_path, e))
+
+    # function to get modelling sheet data
+    def __getModellignSheetData(self, fileName):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:293]: correct the spelling of the function.

Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:293]: make it a generic function that reads data from any sheet. parameter sheetName will "modelling"

File: vevestaX/vevesta.py [hunk 19]
```diff
@@ -201,92 +203,102 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
-        script_dir = os.path.dirname('vevestaXdump')
-        results_dir = os.path.join(script_dir, '/')
-        sample_file_name = column
-
-        if not os.path.isdir(results_dir):
-            os.makedirs(results_dir)
-
-        fig.savefig(results_dir + sample_file_name)
-
     def commit(self, techniqueUsed, filename=None, message=None, version=None, project_id=None):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:302]: Change project_id to projectId through out the code.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py [hunk 20]
```diff
@@ -201,92 +203,102 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+            workBook.create_sheet('performancePlots')
+            plotSheet=workBook['performancePlots']
+            xAxis = list(nonNumericColumns['timestamp'])
+            columnValue = 2
+
+            for column in modelingData.columns:
+                yAxis = list(modelingData[column])
+
+                imageName = str(column)+'.png'
+                columnText = 'A'
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:254]: plot the plots in B column, leave A column

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py (hunk 18 above)
Comment (Priyanka-Vevesta) [reply vevestaX/vevesta.py:293]: call it __getExcelSheetData(self, fileName)

File: vevestaX/vevesta.py [hunk 21]
```diff
@@ -201,93 +204,102 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
...
+                try:
+                    if os.path.isfile(file_path) or os.path.islink(file_path):
+                        os.unlink(file_path)
+                    elif os.path.isdir(file_path):
+                        shutil.rmtree(file_path)
+                except Exception as e:
+                    print('Failed to delete %s. Reason: %s' % (file_path, e))
+
+    # generic function to get modelling sheet data from any excel file with sheetName modelling
+    def getModellingSheetData(self, fileName):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:293]: this is not a generic function use function signature __getExcelSheetData(self, fileName, sheetName)

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

**Review by Priyanka-Vevesta — APPROVED** [review]

**Review by Priyanka-Vevesta — APPROVED** [review]

**Pull request merged**